app/src/main/java/com/android/streetworkapp/model/park/ParkViewModel.kt

```diff
@@ -102,6 +110,30 @@ open class ParkViewModel(private val repository: ParkRepository) : ViewModel() {
   fun updateName(pid: String, name: String) =
       viewModelScope.launch { repository.updateName(pid, name) }
 
+  /** Update the name of the current park using the nominatim API. */
+  fun updateCurrentParkNameNominatim() =
+      viewModelScope.launch {
+
+        // Only true when we have never updated the park name with nominatim to reduce calls to the
+        // API
+        if (_currentPark.value?.name?.contains("Default Park") == true) {
+          nameRepository.convertLocationIdToParkName(
+              _parkLocation.value.id,
+              { name -> _currentPark.value?.let { repository.updateName(it.pid, name) } },
+              {})
+        }
+      }
+
+  /**
+   * Set the new parkLocation
+   *
+   * @param parkLocation: a ParkLocation object
+   */
+  fun setParkLocation(parkLocation: ParkLocation?) {
+    require(parkLocation != null) { "ParkLocation can not be null" }
+    _parkLocation.value = parkLocation
+  }
+
```

Comment:
Since this function is related to `_parkLocation`, I'd like to comment it to keep track. No need to reply.